src/App.tsx

```diff
@@ -1,61 +1,152 @@
+import { useState } from 'react';
 import './App.scss';
+import 'bulma/css/bulma.css';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+
+const getUserById = (userId: number): User | null => {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+};
+
+const usersWithTodo: Todo[] = todosFromServer.map((todo) => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+const getNewId = (todos: Todo[]): number => {
+  return Math.max(...todos.map(todo => todo.id)) + 1;
+};
 
 export const App = () => {
+  const [todos, setTodos] = useState(usersWithTodo);
+  const [newTitle, setNewTitle] = useState('');
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [hasSelectedUserError, setHasSelectedUserError] = useState(false);
+
+  const handleChangeTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setNewTitle(event.target.value);
+    setHasTitleError(false);
+  };
+
+  const handleChangeUser = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectedUserId(+event.target.value);
+    setHasSelectedUserError(false);
+  };
+
+  const resetForm = () => {
+    setNewTitle('');
+    setSelectedUserId(0);
+    setHasSelectedUserError(false);
+    setHasTitleError(false);
+  };
+
+  const addNewTodo = (title: string, userId: number) => {
```

Comment:
Consider renaming `addNewTodo` to `handleAddTodo` to better describe what the function does. 🤔